### PR TITLE
Document and check WCS usage for `to_sky` option in subset-to-region translator; add tests

### PR DESCRIPTION
**Pull request by cshanahan1**
Closing https://github.com/glue-viz/glue-astronomy/pull/96/files as the workaround in there can go directly in Jdaviz, but keeping changes to docstring/tests for the `roi_subset_state_to_region` function. Not including a change log entry for this.

### CODE REVIEW

File: glue_astronomy/translators/regions.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 from glue import __version__ as glue_version
 
 from astropy import units as u
+from astropy.wcs.wcsapi import BaseHighLevelWCS
```
Comment (pllim) [inline glue_astronomy/translators/regions.py:10]:
```suggestion
```

File: glue_astronomy/translators/regions.py [hunk 2]
```diff
@@ -56,9 +57,19 @@ def range_to_rect(data, ori, low, high):
...
+
+    If ``to_sky`` is False, it will return the region in pixel coordinates.
+    If ``to_sky=True``, it will return the region transformed to sky
+    coordinates, per attached data WCS.
+
+    Parameters
+    ----------
+    subset_state : `~glue.core.subset.SubsetState`
+        ROI subset state.
+    to_sky: bool or WCS object
```
Comment (pllim) [inline glue_astronomy/translators/regions.py:70]:
```suggestion
    to_sky: bool
```

File: glue_astronomy/translators/tests/test_regions.py [hunk 3]
```diff
@@ -32,25 +33,37 @@ def setup_method(self, method):
...
 
-        subset_state = RoiSubsetState(self.data.pixel_component_ids[1],
-                                      self.data.pixel_component_ids[0],
-                                      RectangularROI(1, 3.5, -0.2, 3.3))
+        rois = {'CircularROI': RoiSubsetState(self.data.pixel_component_ids[1],
+                                              self.data.pixel_component_ids[0],
+                                              CircularROI(1, 3.5, 0.75)),
+                'RectangularROI': RoiSubsetState(self.data.pixel_component_ids[1],
+                                                 self.data.pixel_component_ids[0],
+                                                 RectangularROI(0, 2, 0, 7)),
```
Comment (pllim) [inline glue_astronomy/translators/tests/test_regions.py:45]: Any reason to change existing test input? I think it is good to test out of bounds coords.

```suggestion
                                                 RectangularROI(1, 3.5, -0.2, 3.3)),
```

File: glue_astronomy/translators/tests/test_regions.py [hunk 4]
```diff
@@ -32,25 +33,37 @@ def setup_method(self, method):
         self.data.coords = WCS_CELESTIAL
         self.dc = DataCollection([self.data])
 
-    def test_rectangular_roi(self):
+    @pytest.fixture
```
Comment (pllim) [inline glue_astronomy/translators/tests/test_regions.py:36]: Fixture is overkill. I think it is enough to define the dict at module level, and then pass the var in using https://docs.pytest.org/en/stable/how-to/parametrize.html in `test_roi_subset_state_to_region`.

As for the rest of the test methods, they might be able to see a module level var without any special setup but do give that a try locally first.

If you insist on fixture anyway, please put it in `conftest.py`, not embedded inside a test class. Jdaviz should have example of this.

File: glue_astronomy/translators/tests/test_regions.py [hunk 5]
```diff
@@ -32,25 +33,37 @@ def setup_method(self, method):
...
+
+        subset_state = setup_rois['RectangularROI']
 
         self.dc.new_subset_group(subset_state=subset_state, label='rectangular')
 
         reg = self.data.get_selection_definition(format='astropy-regions')
 
         assert isinstance(reg, RectanglePixelRegion)
 
-        assert_allclose(reg.center.x, 2.25)
```
Comment (pllim) [inline glue_astronomy/translators/tests/test_regions.py:47]: Shouldn't have changed this.

File: glue_astronomy/translators/tests/test_regions.py [hunk 6]
```diff
@@ -488,3 +497,32 @@ def test_unsupported(self):
...
+
+        # test sky region
+        reg_sky = roi_subset_state_to_region(subset_state, to_sky=True)
+        assert isinstance(reg_sky, output_sky_shape)
+        assert_allclose(reg_sky.center.ra.deg, 1.99918828)
+        assert_allclose(reg_sky.center.dec.deg, 4.48805907)
+
+        # and that proper error is raised when there is no WCS
+        with pytest.raises(ValueError, match="No WCS associated with subset data, can't do to_sky transformation."):
+            subset_state.xatt.parent.coords = None
```
Comment (pllim) [inline glue_astronomy/translators/tests/test_regions.py:527]: This changes in-place something that is used in other tests. I don't think this is safe.

File: glue_astronomy/translators/tests/test_regions.py (hunk 3 above)
Comment (cshanahan1) [reply glue_astronomy/translators/tests/test_regions.py:45]: i wanted the centers of all subsets to be the same so parameterizing the test was simpler.

File: glue_astronomy/translators/tests/test_regions.py (hunk 4 above)
Comment (cshanahan1) [reply glue_astronomy/translators/tests/test_regions.py:36]: done, its not a fixture anymore.

File: glue_astronomy/translators/tests/test_regions.py (hunk 6 above)
Comment (cshanahan1) [reply glue_astronomy/translators/tests/test_regions.py]: fixed.

**Review by pllim — COMMENTED** [review]
Much cleaner now. Thanks!

@dhomeier would need to enable the CI and do a final review.

**Comment by dhomeier** [comment]
@pllim do you know if a `spectral_cube` release with your fix is planned, or should we move that to dev-deps at least for now, to unbreak astropy-dev?

**Comment by pllim** [comment]
spectral-cube release is up to @keflavich or @astrofrog , I think?

**Comment by pllim** [comment]
Maybe it is time to seriously think about:

* #62

File: glue_astronomy/translators/regions.py [hunk 7]
```diff
@@ -56,9 +56,19 @@ def range_to_rect(data, ori, low, high):
...
+
+    If ``to_sky`` is False, it will return the region in pixel coordinates.
+    If ``to_sky=True``, it will return the region transformed to sky
+    coordinates, per attached data WCS.
+
+    Parameters
+    ----------
+    subset_state : `~glue.core.subset.SubsetState`
+        ROI subset state.
+    to_sky: bool
```
Comment (dhomeier) [inline glue_astronomy/translators/regions.py:69]:
```suggestion
    to_sky : bool
```

**Comment by e-koch** [comment]
@dhomeier does radio-beam also just need a new release? Or is there another issue?

**Comment by dhomeier** [comment]
Having all imported in sync from the dev branches in #98 resolves the issue now; it's probably more problematic for downstream packages that are not explicitly testing spectral_cube and deps.

**Comment by keflavich** [comment]
we pushed a new release.  If either of you would like to join on as maintainers to help with the release cycle, let us know.

**Review by dhomeier — APPROVED** [review]

**Pull request merged**